source/common/http/conn_manager_impl.cc

```diff
@@ -2004,6 +2014,12 @@ bool ConnectionManagerImpl::ActiveStream::createFilterChain() {
   }
   bool upgrade_rejected = false;
   auto upgrade = request_headers_ ? request_headers_->Upgrade() : nullptr;
```

Comment:
nit: can you make this a non-auto type? It's a little confusing to read through and see what type this is.